src/main/java/org/opentripplanner/analyst/core/WeightingFunction.java

```diff
@@ -0,0 +1,93 @@
+package org.opentripplanner.analyst.core;
+
+/**
+ * A weighting function, expressing how influential something is according to its distance from you.
+ * A functional interface (with a single function, so it can be defined using Java 8 shorthand).
+ */
+public abstract class WeightingFunction {
+
+    public abstract double getWeight (int seconds);
```

Comment:
Hmm, I guess my statement in the Javadoc about this being a valid @FunctionalInterface is not true since there are two functions (and it's not an interface). The separation into two functions was to allow optimizations for e.g. hard-cutoff functions, where you can avoid doing a multiplication to determine the final result. I'm not sure that actually has much of an effect on execution speed, maybe the functional abstraction is more important. But maybe we'll always just use the same function, in which case the abstraction is purely to aid understanding.